### PR TITLE
[ios] Set contentsScale before we commit CATransaction

### PR DESCRIPTION
**Pull request by iskakaushik**
Layout occurs after `[CATransaction commit]`. `layoutSubviews` was where we set the `contentsScale` on the `CALayer`. This meant that for one frame, we would see content on the overlay view which was did not have the correct content scale.

This change makes it so that we initialize the `FlutterOverlayView` with the correct `contentsScale`.

This also updates the `overlay_gr_context_` when we first create the `overlay_view`. This is an artifact of https://github.com/flutter/engine/pull/8175.

This manifests as jank as seen in: https://github.com/flutter/flutter/issues/29573

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterOverlayView.h [hunk 1]
```diff
@@ -23,6 +23,7 @@
 - (instancetype)initWithCoder:(NSCoder*)aDecoder NS_UNAVAILABLE;
 
 - (instancetype)init NS_DESIGNATED_INITIALIZER;
+- (instancetype)initForGLWithContentsScale:(CGFloat)contentsScale;
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/FlutterOverlayView.h]: Just say `initWithContentsScale` and check the layer class in the implementation before setting the content and rasterization scales.

File: shell/platform/darwin/ios/framework/Source/FlutterOverlayView.mm [hunk 2]
```diff
@@ -43,15 +43,20 @@ - (instancetype)init {
...
-    layer.contentsScale = screenScale;
-    layer.rasterizationScale = screenScale;
+    layer.contentsScale = contentsScale;
+    layer.rasterizationScale = contentsScale;
   }
 
+  return self;
+}
+
+- (void)layoutSubviews {
```
Comment (iskakaushik) [inline shell/platform/darwin/ios/framework/Source/FlutterOverlayView.mm]: @chinmaygarde do I still need to override this method?

Comment (iskakaushik) [reply shell/platform/darwin/ios/framework/Source/FlutterOverlayView.mm]: i guess not

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 3]
```diff
@@ -286,6 +287,7 @@
   std::unique_ptr<Surface> surface = ios_surface->CreateSecondaryGPUSurface(gr_context);
   overlays_[overlay_id] = std::make_unique<FlutterPlatformViewLayer>(
       fml::scoped_nsobject<UIView>(overlay_view), std::move(ios_surface), std::move(surface));
+  overlays_gr_context_ = gr_context;
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: mention this in the PR description

**Pull request merged**